### PR TITLE
[MBP-381] Extract the GitopsConfig spec from the Pattern CRD

### PR DESCRIPTION
**Pull request by jordigilh**
This PR extracts the contents of the gitopsconfig spec in the pattern CRD that are defined the configuration of the openshift-gitops subscription and changes the code so that the controller will instead check at bootup time for a configmap in its own namespace named `patterns-operator-gitops-config` for a custom configuration of the gitops operator.
The configmap should contain a file named `gitops-config.json`. Although the contents of the file match the OLM's subscription spec schema, the controller will only monitor for these fields:
* Channel
* CatalogSource
* StartingCSV
* InstallPlanApproval

These are the same fields targeted by the current implementation when defining a custom gitops configuration. 

@mbaldessari @beekhof PTAL

### CODE REVIEW

File: bundle/manifests/patterns-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -295,10 +278,19 @@ spec:
...
+                - mountPath: /gitops-config/
+                  name: gitops-config
+                  readOnly: true
               securityContext:
                 runAsNonRoot: true
               serviceAccountName: patterns-operator-controller-manager
               terminationGracePeriodSeconds: 10
+              volumes:
+              - configMap:
+                  name: patterns-operator-gitops-config
```
Comment (beekhof) [inline bundle/manifests/patterns-operator.clusterserviceversion.yaml:291]: what's the failure mode if the config map doesn't exist?
I don't see one being created by default

File: controllers/pattern_controller.go [hunk 2]
```diff
@@ -469,11 +450,42 @@ func (r *PatternReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 	r.driftWatcher, _ = newDriftWatcher(r.Client, mgr.GetLogger(), newGitClient())
+
+	if err = r.deployGitopsSubscription(customConfigFile); err != nil {
+		return err
+	}
```
Comment (beekhof) [inline controllers/pattern_controller.go:456]: This isn't something we should be doing during setup

File: bundle/manifests/patterns-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (jordigilh) [reply bundle/manifests/patterns-operator.clusterserviceversion.yaml:291]: It's optional at this point to make sure it's backwards compatible.

File: controllers/pattern_controller.go (hunk 2 above)
Comment (jordigilh) [reply controllers/pattern_controller.go:456]: ack

**Pull request merged**